### ISSUE TITLE
[ci] [R-package] macOS CI jobs failing: "Download failed on Cask 'basictex'"

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#6366]
## Description

macOS R-package CI jobs here started failing some time in the last few days, with errors like this:

```text
==> Downloading https://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20230314.pkg
curl: (22) The requested URL returned error: 404
Error: Download failed on Cask 'basictex' with message: Download failed: https://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20230314.pkg
Error: Process completed with exit code 1.
```

## Reproducible example

All macOS R-package jobs in this project. Example build: ([build link](https://github.com/microsoft/LightGBM/actions/runs/8309555610/job/22744122077?pr=6357))

## Additional Comments

I believe this is happening right here:

https://github.com/microsoft/LightGBM/blob/b27d81ea411d04d8d071d4d4e75c19ffa15c5795/.ci/test_r_package.sh#L90

`basictex`'s formula was updated about a week ago, to version `2024.0309`:

* `basictex`: https://github.com/Homebrew/homebrew-cask/pull/168996

and `mactex`'s to `2024.0312`

* `mactex`: https://github.com/Homebrew/homebrew-cask/pull/168992
* `mactex-no-gui`: https://github.com/Homebrew/homebrew-cask/pull/168993

I suspect that mirrors have been updated to include those new packages and that the old ones have been removed, but Homebrew in CI is still looking for the old versions.

Maybe a `brew update` will help, as described in these:

* https://github.com/Homebrew/homebrew-cask/issues/143642#issuecomment-1485259149
* #5925
* #5807
* #4161

**Comment by jameslamb** [comment]
Seems that just adding the following on macOS jobs in #6357 fixed this:

```shell
brew update-reset
brew update
```

Think that's evidence that the issue was "there was a new basictex release and Homebrew in GitHub Actions' VMs just needed to be updated to look for it".

**Comment by jameslamb** [comment]
This was resolved by #6357
